src/main_loop/proof_upgrader.rs

```diff
@@ -422,6 +426,39 @@ impl UpgradeJob {
         }
     }
 
+    fn gobbler_notification_method_with_receiver_preimage(
```

Comment:
Non-public functions don't belong in the middle of a list of public function declarations. They belong either at the bottom or at the top of the module. Or, if it's not tested, it can also be declared as a helper function inside the function where it's used.